### DISCUSSION TITLE
Can we use other MQTT enabled sensors to drive faikin auto mode?

### DISCUSSION THREAD

**Discussion by mmmgggttt** [revk/xESP32-Faikout#383] [General]
Hello, i do have TRVs working on zigbee2mqtt which are used in winter to drive my heaters. They do , during summer , provide info on temperature. Can I integrate this to allow faikin to drive the auto mode?

**Comment by revk** [comment]
We handle several types of BLE sensors. If it shows on the Faikin when you enable BLE, great, if not, probably not. It does not talk zigbee.

**Comment by mmmgggttt** [comment]
i didnt mean to use zigbee indeed.
What i'm saying that us tech savvy are probably having a bunch of sensors for any reasons already present in our homes for different purposes.
In my case i know ambient temps of different rooms also in summer, due to the presence of heaters TRVs (modulating valve for radiator heaters for the winter) that are connected on the zigbee network to my HA. 
If i would be able to "inject" the ambient temperature reading to FAIKIN throughout MQTT and HA , i would then be able to set auto mode for a remote room , without having to buy a BLE sensor.

I hope this make sense.

maybe it's already possible playing around with MQTT and HA but not sure.

thank you

**Reply by revk** [reply]
Ah I see, yes, you can send regular messages to Faikin to advise current reference temperature, as per the Manual.

Sending a `control` message with `env`.

<img width="634" alt="Screenshot 2024-07-10 at 09 51 03" src="https://github.com/revk/ESP32-Faikin/assets/996983/0d74b41c-f6e5-4b39-871b-dda2a309e942">

No idea how to make HA send that, but that is how Faikin can be driven remotely.

**Reply by mmmgggttt** [reply]
ok thank you , but is it possible having the faikin listening to a specific topic ? In my case ...
zigbee2mqtt/TRV_salacentrale
which is publishing the following 
{"ADC":"1215,1216,0,284,-32768,2000,0,0,100,284,284,284","ALG":"1407,1,200,284,-84,0,0,0,0,0,100","MOT":"CloseValve","keypad_lockout":"unlock","linkquality":54,"local_temperature":28.4,"occupied_heating_setpoint":20,"pi_heating_demand":0,"running_state":null,"voltage":2900}

and wishing to use the value in field local temperature to use it as ENV ....how to achieve this ?

**Reply by revk** [reply]
So you want a configurable topic and JSON extraction from payload for reference temperature.

I mean, not impossible. I'll have top think about it.

**Reply by mmmgggttt** [reply]
yeah , i think we either have the possibility to faikin to listen to a topic.... OR ...i guess, trough HA  and node red taking the message from zigbee2mqtt/TRV_salacentrale , rewriting a new message to faikin with the extracted value , and publish it...

**Reply by revk** [reply]
It does sound more like something HA would be good at, to be honest.
